File: LeetCode/Easy/1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int count=0, index=0;
        if(nums.size()==1){
            return true;
        }
        for(int i=0;i<nums.size()-1;i++){
            if(nums[i]>nums[i+1]){
                count++;
                index=i+1;
            }
        }
        if(count>1){
            return false;
        }
        else if(count==1){
            if(nums[nums.size()-1]<=nums[0]){
                return true;
            }
            else{
                return false;
            }
        }
        return true;
    }
};

File: LeetCode/Easy/1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int n=nums.length;
        int[] copy=new int[n];
        for(int i=0;i<n;i++) {
           copy[i]=nums[i];
        }
        Arrays.sort(copy);
        int index=0;
        for(int i=0;i<n;i++) {
            if(nums[i]==copy[0] && (i-1)>=0 && nums[i]<nums[i-1]) {
                index=i;
                break;
            }
        }
        for(int i=0;i<n;i++) {
            if(copy[i]!=nums[(i+index)%n]) {
                return false;
            }
        }
        return true;
    }
}